appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -1830,67 +2313,72 @@ private void processConditionalAnnotations(ComponentInfo componentInfo, Element
   protected abstract void outputResults() throws IOException;
 
   /**
-   * Returns the appropriate Yail type (e.g., "number" or "text") for a
-   * given Java type (e.g., "float" or "java.lang.String").  All component
-   * names are converted to "component".
+   * Returns the appropriate Yail type for a given Java type.
    *
-   * @param type a type name, as returned by {@link TypeMirror#toString()}
-   * @return one of "boolean", "text", "number", "list", or "component".
+   * @param type a TypeMirror representing the Java type.
+   * @return the equivalent Yail type. All component names are converted to "component".
    * @throws RuntimeException if the parameter cannot be mapped to any of the
    *         legal return values
    */
-  protected final String javaTypeToYailType(String type) {
+  protected final String javaTypeToYailType(TypeMirror type) {
     if (BOXED_TYPES.containsKey(type)) {
       throw new IllegalArgumentException(String.format(BOXED_TYPE_ERROR, type,
           BOXED_TYPES.get(type)));
     }
+
+    // Handle enums
+    if (isOptionList(type)) {
+      return type.toString() + "Enum";
```

Comment:
Justify this line with a comment. I.e., in YAIL code generation, you will want an easy test to determine which types represent abstract options. You have chosen to do that by having every such type end with "Enum"

Reply:
Good idea! [Done](https://github.com/BeksOmega/appinventor-sources/pull/3/files#diff-6974a53a12838239a1c808e17a441cd2R2365).